### DISCUSSION TITLE
API : Way to load JSON files directly into ControlNet?

### DISCUSSION THREAD

**Discussion by CoolCuda** [Mikubill/sd-webui-controlnet#2943] [Q&A]
Hello,

Do you know how to load Json file instead of image file into the api of controlnet and automatic1111 ?

My script : 
$data['alwayson_scripts'] = array(
        'controlnet' => array(
            'args' => array(
                array(
                    'input_image' => $imagecontrolnetData,
                    ...
                )
            )

It can be cool if we have instead of    'input_image'  something like that : 
                  'input_json' => $my_json_file

because .json are more accurate than image...
 
Thank you :)

**Comment by huchenlei** [comment]
See https://github.com/Mikubill/sd-webui-controlnet/discussions/2854#discussioncomment-9335880 for extensions that load ControlNet unit state.

We have no intention implementing a way to directly load JSON file.

**Comment by CoolCuda** [comment]
No problem, Thank you for your reply :)
